app/assets/javascripts/posts/editor.js

```diff
@@ -152,7 +152,8 @@ function setupWritableEditor() {
     // Set the ID
     var id = $(this).val();
     $("#reply_character_alias_id").val(id);
-    $("#post-editor .post-character #name").html($('#character_alias option:selected').text());
+    var correctID = $("input#reply_character_id")[0].value
+    $("#post-editor .post-character #name").html(`<a href="/characters/${correctID}">${$('#character_alias option:selected').text()}</a>`);
```

Comment:
Per discussion in Discord these would be safer as JQuery nodes, e.g. `$("<a>").text(name)` etc

Reply:
```suggestion
    var nameNode = $("<a>").attr("href", "/characters/"+correctID).text($('#character_alias option:selected').text());
    $("#post-editor .post-character #name").html(nameNode);
```